File: Chapter 3/3.2/uva/11130.cpp
#include<cstdio>
#include<cmath>

using namespace std;

int main(){
    long long a, b, xa, xb;
    long double v, A, s;
    long double xv, xA, xs;
    constexpr double pi = 3.14159265358979323846;

    scanf("%lld %lld %llf %llf %llf\n",&a,&b,&v,&A,&s);
    while(scanf("%lld %lld %llf %llf %llf\n",&xa,&xb,&xv,&xA,&xs)!=EOF){
    
        const long double totalDist {v*s/2.0};

        long dX = long((a/2.0 + totalDist * cos(pi * A/180.0))/a);
        long dY = long((b/2.0 + totalDist * sin(pi * A/180.0))/b);

        printf("%ld %ld\n",dX,dY);
        
        a = xa;
        b = xb;
        v = xv;
        A = xA;
        s = xs;
    }
}

File: Chapter 3/3.2/uva/11130.py
import sys
import math

allInputs = sys.stdin.read().splitlines()[:-1]

for curr in allInputs:
    values = list(map(int,curr.split(" ")))

    a = values[0]
    b = values[1]
    v = values[2]
    A = values[3]
    s = values[4]

    totalDist = v*s/2

    dX = int(a/2 + totalDist * math.cos(math.pi * A/180))//a
    dY = int(b/2 + totalDist * math.sin(math.pi * A/180))//b

    print(dX, dY)